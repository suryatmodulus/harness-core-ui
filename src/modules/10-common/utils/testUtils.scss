/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
