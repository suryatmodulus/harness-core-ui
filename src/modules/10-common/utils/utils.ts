/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
