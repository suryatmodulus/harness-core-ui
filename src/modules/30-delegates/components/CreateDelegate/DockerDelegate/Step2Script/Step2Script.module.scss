/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.dockerDetailsContainer {
  padding-left: var(--spacing-xxlarge) !important;
  flex: 1;
}

.verificationField {
  box-shadow: 0px 0px 5px rgba(40, 41, 61, 0.0348285);
  background: #effbff !important;
  border-radius: 4px !important;
}

.verticalDivider {
  margin: 0 var(--spacing-large) !important;
}
