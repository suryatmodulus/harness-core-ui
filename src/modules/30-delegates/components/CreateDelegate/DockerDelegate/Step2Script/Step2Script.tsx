/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useEffect, useState } from 'react'
import { useParams } from 'react-router-dom'
import set from 'lodash-es/set'
import { Button, Layout, StepProps, Heading, Text, Color, Container } from '@wings-software/uicore'

import CopyToClipboard from '@common/components/CopyToClipBoard/CopyToClipBoard'
import { useStrings } from 'framework/strings'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import YamlBuilder from '@common/components/YAMLBuilder/YamlBuilder'

import { useGenerateDockerDelegateYAML, DelegateSetupDetails } from 'services/portal'

import type { DockerDelegateWizardData } from '../CreateDockerDelegate'
import css from './Step2Script.module.scss'

const dockerFileName = 'docker-compose.yaml'
const dockerComposeCommand = 'docker-compose -f docker-compose.yaml up -d'

const getCommandBlock = (label: string, command: string) => (
  <>
    <Text margin={{ top: 'medium' }} color={Color.BLACK}>
      {label}
    </Text>
    <Layout.Horizontal spacing="medium">
      <Container
        intent="primary"
        padding="small"
        font={{
          align: 'center'
        }}
        flex
        className={css.verificationField}
      >
        <Text margin={{ right: 'xxlarge' }} font="small">
          {`$ ${command}`}
        </Text>
        <CopyToClipboard content={command} />
      </Container>
    </Layout.Horizontal>
  </>
)

const Step2Script: React.FC<StepProps<DockerDelegateWizardData>> = props => {
  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
  const [delegateDockerYaml, setDelegateDockerYaml] = useState('')
  const { getString } = useStrings()
  const { previousStep, prevStepData } = props

  const linkRef = React.useRef<HTMLAnchorElement>(null)

  const { mutate: getDockerYaml } = useGenerateDockerDelegateYAML({
    queryParams: {
      accountId
    }
  })

  const fetchDockerYaml = async () => {
    const createParams = { ...prevStepData } as DelegateSetupDetails

    if (projectIdentifier) {
      set(createParams, 'projectIdentifier', projectIdentifier)
    }
    if (orgIdentifier) {
      set(createParams, 'orgIdentifier', orgIdentifier)
    }
    set(createParams, 'delegateType', 'DOCKER')
    const dockerYaml = (await getDockerYaml(createParams)) as any
    setDelegateDockerYaml(dockerYaml)
  }

  useEffect(() => {
    fetchDockerYaml()
  }, [])

  const onDownload = () => {
    const content = new Blob([delegateDockerYaml as BlobPart], { type: 'data:text/plain;charset=utf-8' })
    if (linkRef?.current) {
      linkRef.current.href = window.URL.createObjectURL(content)
      linkRef.current.download = dockerFileName
      linkRef.current.click()
    }
  }

  const yamlBuilderComp = (
    <YamlBuilder
      entityType="Delegates"
      fileName={dockerFileName}
      isReadOnlyMode={true}
      isEditModeSupported={false}
      existingYaml={delegateDockerYaml as any}
      showSnippetSection={false}
      width="568px"
      height="462px"
      theme="DARK"
    />
  )

  return (
    <>
      <Layout.Horizontal padding={{ top: 'small', left: 'xlarge', right: 'xlarge' }}>
        <Layout.Vertical spacing="medium">
          <div>{yamlBuilderComp}</div>
          <Layout.Horizontal padding={{ top: 'medium' }}>
            <Button
              id="stepReviewScriptDownloadButton"
              icon="arrow-down"
              text={getString('delegates.downloadYAMLFile')}
              onClick={onDownload}
              outlined
            />
          </Layout.Horizontal>
          <Layout.Horizontal padding={{ top: 'medium' }} spacing="small">
            <Button
              id="stepReviewScriptBackButton"
              text={getString('back')}
              icon="chevron-left"
              onClick={() => previousStep?.({ ...prevStepData })}
              intent="none"
            />
            <Button
              id="stepReviewScriptContinueButton"
              type="submit"
              intent="primary"
              text={getString('continue')}
              rightIcon="chevron-right"
              onClick={() => {
                props.nextStep?.({ ...prevStepData })
              }}
            />
          </Layout.Horizontal>
        </Layout.Vertical>
        <Layout.Vertical className={css.verticalDivider}>
          <hr />
        </Layout.Vertical>
        <Layout.Vertical>
          <Heading level={2} margin={{ bottom: 'large' }} color={Color.BLACK}>
            {getString('delegates.delegateCreation.docker.verifyTitle')}
          </Heading>

          <Text lineClamp={2} font="normal" margin={{ bottom: 'medium' }}>
            {getString('delegates.delegateCreation.docker.verifyDesc1')}
          </Text>

          <Text lineClamp={2} margin={{ bottom: 'medium' }}>
            {getString('delegates.delegateCreation.docker.verifyDesc2')}
          </Text>

          <Layout.Horizontal spacing="medium">
            <Container
              intent="primary"
              padding="small"
              font={{
                align: 'center'
              }}
              flex
              className={css.verificationField}
            >
              <Text margin={{ right: 'xxlarge' }} font="small">
                {`$ ${dockerComposeCommand}`}
              </Text>
              <CopyToClipboard content={dockerComposeCommand} />
            </Container>
          </Layout.Horizontal>
        </Layout.Vertical>
        <div className={css.dockerDetailsContainer} style={{ display: 'none' }}>
          <Heading level={2} font={{ weight: 'bold' }} color={Color.BLACK} margin={{ bottom: 'large' }}>
            {getString('delegates.delegateCreation.docker.scriptTitle')}
          </Heading>

          {getCommandBlock(getString('delegates.delegateCreation.docker.scriptText'), '')}

          {getCommandBlock(getString('delegates.delegateCreation.docker.scriptCommandGetIds'), 'docker ps')}

          {getCommandBlock(
            getString('delegates.delegateCreation.docker.scriptSeeLogs'),
            'docker logs -f <container-id>'
          )}

          {getCommandBlock(
            getString('delegates.delegateCreation.docker.scriptRunInShell'),
            'docker container exec -it <container-id> bash'
          )}

          <Text margin={{ top: 'xlarge' }} color={Color.BLACK}>
            {getString('delegates.delegateCreation.docker.docLinkBefore')}
            <a
              rel="noreferrer"
              href="https://ngdocs.harness.io/article/5ww21ewdt8-configure-delegate-proxy-settings"
              target="_blank"
            >
              {getString('delegates.delegateCreation.docker.docLink')}
            </a>
          </Text>
        </div>
      </Layout.Horizontal>
      <a className="hide" ref={linkRef} target={'_blank'} />
    </>
  )
}

export default Step2Script
