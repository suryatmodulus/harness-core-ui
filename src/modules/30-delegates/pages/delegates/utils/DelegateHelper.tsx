/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { useStrings } from 'framework/strings'
import { DelegateTypes } from '@delegates/constants'

export const GetDelegateTitleTextByType = (type: string): string => {
  const { getString } = useStrings()

  switch (type) {
    case DelegateTypes.KUBERNETES_CLUSTER:
      return getString('kubernetesText')
    default:
      /* istanbul ignore next */
      return ''
  }
}
