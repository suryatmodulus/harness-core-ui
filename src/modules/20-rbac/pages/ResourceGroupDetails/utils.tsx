import produce from 'immer'
import { RbacResourceGroupTypes } from '@rbac/constants/utils'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import { isDynamicResourceSelector } from '@rbac/utils/utils'
import type { ResourceSelector } from 'services/resourcegroups'

export const getSelectedResourcesMap = (
  resourceSelectorList?: ResourceSelector[]
): Map<ResourceType, string[] | string> => {
  const map: Map<ResourceType, string[] | string> = new Map()

  resourceSelectorList?.map(resourceSelector => {
    if (isDynamicResourceSelector(resourceSelector.type)) {
      map.set(resourceSelector.resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
    } else {
      map.set(resourceSelector.resourceType, resourceSelector.identifiers)
    }
  })
  return map
}

export const validateResourceSelectors = (value: ResourceSelector[]): ResourceType[] => {
  return value
    .filter(val => {
      if (!isDynamicResourceSelector(val.type)) {
        if (val.identifiers?.length < 1) {
          return true
        }
      }
      return false
    })
    .map(val => val.resourceType)
}

export const getResourceSelectorsfromMap = (map: Map<ResourceType, string[] | string>): ResourceSelector[] => {
  const resourceSelectors: ResourceSelector[] = []
  map.forEach((value, key) => {
    if (isDynamicResourceSelector(value)) {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR,
        resourceType: key
      })
    } else {
      resourceSelectors.push({
        type: RbacResourceGroupTypes.STATIC_RESOURCE_SELECTOR,
        resourceType: key,
        identifiers: value
      })
    }
  })
  return resourceSelectors
}

export const computeResourceMapOnChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  resourceType: ResourceType,
  isAdd: boolean,
  identifiers?: string[]
): void => {
  if (identifiers) {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, identifiers)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          if (Array.isArray(draft.get(resourceType))) {
            draft.set(
              resourceType,
              (draft.get(resourceType) as string[]).filter(el => !identifiers.includes(el))
            )
          }
          if (draft.get(resourceType)?.length === 0) {
            draft.delete(resourceType)
          }
        })
      )
    }
  } else {
    if (isAdd) {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      )
    } else {
      setSelectedResourceMap(
        produce(selectedResourcesMap, draft => {
          draft.delete(resourceType)
        })
      )
    }
  }
}

export const computeResourceMapOnMultiChange = (
  setSelectedResourceMap: (value: React.SetStateAction<Map<ResourceType, string | string[]>>) => void,
  selectedResourcesMap: Map<ResourceType, string | string[]>,
  types: ResourceType[],
  isAdd: boolean
): void => {
  if (isAdd) {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.set(resourceType, RbacResourceGroupTypes.DYNAMIC_RESOURCE_SELECTOR)
        })
      })
    )
  } else {
    setSelectedResourceMap(
      produce(selectedResourcesMap, draft => {
        types.forEach(resourceType => {
          draft.delete(resourceType)
        })
      })
    )
  }
}
