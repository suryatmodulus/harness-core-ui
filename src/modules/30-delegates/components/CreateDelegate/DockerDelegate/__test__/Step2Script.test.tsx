/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { TestWrapper } from '@common/utils/testUtils'
import Step2Script from '../Step2Script/Step2Script'

jest.mock('services/portal', () => ({
  useGenerateDockerDelegateYAML: jest.fn().mockImplementation(() => {
    return {
      mutate: jest.fn().mockImplementation(() => '')
    }
  })
}))

describe('Create Docker Step2Script', () => {
  test('render component and go back', async () => {
    const { container, getByRole } = render(
      <TestWrapper pathParams={{ accountId: 'testAcc', orgIdentifier: 'org1', projectIdentifier: 'proj1' }}>
        <Step2Script />
      </TestWrapper>
    )
    const backBtn = getByRole('button', { name: /back/ })

    userEvent.click(backBtn!)

    expect(container).toMatchSnapshot()
  })

  test('render component and go next', async () => {
    const { getByRole } = render(
      <TestWrapper pathParams={{ accountId: 'testAcc', orgIdentifier: 'org1', projectIdentifier: 'proj1' }}>
        <Step2Script />
      </TestWrapper>
    )
    const continueBtn = getByRole('button', { name: /continue/ })
    userEvent.click(continueBtn!)
    expect(continueBtn).toBeDisabled()
  })
})
