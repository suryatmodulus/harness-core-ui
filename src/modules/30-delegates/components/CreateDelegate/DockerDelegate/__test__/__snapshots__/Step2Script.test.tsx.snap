// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Docker Step2Script render component and go back 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main StyledProps--padding-top-small StyledProps--padding-left-xlarge StyledProps--padding-right-xlarge"
  >
    <div
      class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
    >
      <div>
        <div />
      </div>
      <div
        class="Layout--horizontal StyledProps--main StyledProps--padding-top-medium"
      >
        <button
          class="bp3-button bp3-outlined Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon"
          id="stepReviewScriptDownloadButton"
          type="button"
        >
          <span
            class="bp3-icon bp3-icon-arrow-down StyledProps--main StyledProps--padding-right-xsmall"
            icon="arrow-down"
          >
            <svg
              data-icon="arrow-down"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                arrow-down
              </desc>
              <path
                d="M13 8c-.3 0-.5.1-.7.3L9 11.6V2c0-.5-.4-1-1-1s-1 .5-1 1v9.6L3.7 8.3C3.5 8.1 3.3 8 3 8c-.5 0-1 .5-1 1 0 .3.1.5.3.7l5 5c.2.2.4.3.7.3s.5-.1.7-.3l5-5c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            delegates.downloadYAMLFile
          </span>
        </button>
      </div>
      <div
        class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--padding-top-medium"
      >
        <button
          class="bp3-button bp3-disabled bp3-loading Button--button StyledProps--font StyledProps--main Button--with-current-color Button--withLeftIcon"
          disabled=""
          id="stepReviewScriptBackButton"
          tabindex="-1"
          type="button"
        >
          <div
            class="bp3-spinner bp3-button-spinner"
          >
            <div
              class="bp3-spinner-animation"
            >
              <svg
                height="20"
                stroke-width="16.00"
                viewBox="-3.00 -3.00 106.00 106.00"
                width="20"
              >
                <path
                  class="bp3-spinner-track"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                />
                <path
                  class="bp3-spinner-head"
                  d="M 50,50 m 0,-45 a 45,45 0 1 1 0,90 a 45,45 0 1 1 0,-90"
                  pathLength="280"
                  stroke-dasharray="280 280"
                  stroke-dashoffset="210"
                />
              </svg>
            </div>
          </div>
          <span
            class="bp3-icon bp3-icon-chevron-left StyledProps--main StyledProps--padding-right-xsmall"
            icon="chevron-left"
          >
            <svg
              data-icon="chevron-left"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                chevron-left
              </desc>
              <path
                d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
          <span
            class="bp3-button-text"
          >
            back
          </span>
        </button>
        <button
          class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--withRightIcon"
          id="stepReviewScriptContinueButton"
          type="submit"
        >
          <span
            class="bp3-button-text"
          >
            continue
          </span>
          <span
            class="bp3-icon bp3-icon-chevron-right StyledProps--main"
            icon="chevron-right"
          >
            <svg
              data-icon="chevron-right"
              height="16"
              viewBox="0 0 16 16"
              width="16"
            >
              <desc>
                chevron-right
              </desc>
              <path
                d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="Layout--vertical StyledProps--main verticalDivider"
    >
      <hr />
    </div>
    <div
      class="Layout--vertical StyledProps--main"
    >
      <h2
        class="StyledProps--font StyledProps--font-h2 StyledProps--main StyledProps--margin-bottom-large StyledProps--color StyledProps--color-black"
        level="2"
      >
        delegates.delegateCreation.docker.verifyTitle
      </h2>
      <p
        class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-normal StyledProps--margin-bottom-medium"
        style="--text-line-clamp: 2;"
      >
        delegates.delegateCreation.docker.verifyDesc1
      </p>
      <p
        class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--margin-bottom-medium"
        style="--text-line-clamp: 2;"
      >
        delegates.delegateCreation.docker.verifyDesc2
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
      >
        <div
          class="StyledProps--font StyledProps--main verificationField StyledProps--intent-primary StyledProps--padding-small StyledProps--font-align-center StyledProps--flex"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--margin-right-xxlarge StyledProps--font-small"
          >
            $ docker-compose -f docker-compose.yaml up -d
          </p>
          <span
            class="bp3-popover-wrapper"
          >
            <span
              class="bp3-popover-target"
            >
              <div
                class=""
                tabindex="0"
              >
                <span
                  class="bp3-icon StyledProps--main copyIcon"
                  data-icon="copy-alt"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 8 9"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M7.273 1.403c0-.109-.073-.181-.182-.181H3.453c-.108 0-.181.072-.181.18v.181a.363.363 0 01-.727 0v-.361c0-.397.328-.722.728-.722h4c.4 0 .727.325.728.721v5.42a.726.726 0 01-.728.722h-1.09a.364.364 0 01-.365-.36c0-.2.165-.362.364-.362h.91c.108 0 .18-.072.18-.18V1.403zM0 7.847V2.958c0-.359.337-.652.748-.652h4.114c.411 0 .749.293.749.652v4.89c0 .359-.338.652-.749.652H.748C.337 8.5 0 8.207 0 7.847z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="dockerDetailsContainer"
      style="display: none;"
    >
      <h2
        class="StyledProps--font StyledProps--font-h2 StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-black StyledProps--margin-bottom-large"
        level="2"
      >
        delegates.delegateCreation.docker.scriptTitle
      </h2>
      <p
        class="StyledProps--font StyledProps--main StyledProps--margin-top-medium StyledProps--color StyledProps--color-black"
      >
        delegates.delegateCreation.docker.scriptText
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
      >
        <div
          class="StyledProps--font StyledProps--main verificationField StyledProps--intent-primary StyledProps--padding-small StyledProps--font-align-center StyledProps--flex"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--margin-right-xxlarge StyledProps--font-small"
          >
            $ 
          </p>
          <span
            class="bp3-popover-wrapper"
          >
            <span
              class="bp3-popover-target"
            >
              <div
                class=""
                tabindex="0"
              >
                <span
                  class="bp3-icon StyledProps--main copyIcon"
                  data-icon="copy-alt"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 8 9"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M7.273 1.403c0-.109-.073-.181-.182-.181H3.453c-.108 0-.181.072-.181.18v.181a.363.363 0 01-.727 0v-.361c0-.397.328-.722.728-.722h4c.4 0 .727.325.728.721v5.42a.726.726 0 01-.728.722h-1.09a.364.364 0 01-.365-.36c0-.2.165-.362.364-.362h.91c.108 0 .18-.072.18-.18V1.403zM0 7.847V2.958c0-.359.337-.652.748-.652h4.114c.411 0 .749.293.749.652v4.89c0 .359-.338.652-.749.652H.748C.337 8.5 0 8.207 0 7.847z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
            </span>
          </span>
        </div>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--margin-top-medium StyledProps--color StyledProps--color-black"
      >
        delegates.delegateCreation.docker.scriptCommandGetIds
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
      >
        <div
          class="StyledProps--font StyledProps--main verificationField StyledProps--intent-primary StyledProps--padding-small StyledProps--font-align-center StyledProps--flex"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--margin-right-xxlarge StyledProps--font-small"
          >
            $ docker ps
          </p>
          <span
            class="bp3-popover-wrapper"
          >
            <span
              class="bp3-popover-target"
            >
              <div
                class=""
                tabindex="0"
              >
                <span
                  class="bp3-icon StyledProps--main copyIcon"
                  data-icon="copy-alt"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 8 9"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M7.273 1.403c0-.109-.073-.181-.182-.181H3.453c-.108 0-.181.072-.181.18v.181a.363.363 0 01-.727 0v-.361c0-.397.328-.722.728-.722h4c.4 0 .727.325.728.721v5.42a.726.726 0 01-.728.722h-1.09a.364.364 0 01-.365-.36c0-.2.165-.362.364-.362h.91c.108 0 .18-.072.18-.18V1.403zM0 7.847V2.958c0-.359.337-.652.748-.652h4.114c.411 0 .749.293.749.652v4.89c0 .359-.338.652-.749.652H.748C.337 8.5 0 8.207 0 7.847z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
            </span>
          </span>
        </div>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--margin-top-medium StyledProps--color StyledProps--color-black"
      >
        delegates.delegateCreation.docker.scriptSeeLogs
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
      >
        <div
          class="StyledProps--font StyledProps--main verificationField StyledProps--intent-primary StyledProps--padding-small StyledProps--font-align-center StyledProps--flex"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--margin-right-xxlarge StyledProps--font-small"
          >
            $ docker logs -f &lt;container-id&gt;
          </p>
          <span
            class="bp3-popover-wrapper"
          >
            <span
              class="bp3-popover-target"
            >
              <div
                class=""
                tabindex="0"
              >
                <span
                  class="bp3-icon StyledProps--main copyIcon"
                  data-icon="copy-alt"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 8 9"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M7.273 1.403c0-.109-.073-.181-.182-.181H3.453c-.108 0-.181.072-.181.18v.181a.363.363 0 01-.727 0v-.361c0-.397.328-.722.728-.722h4c.4 0 .727.325.728.721v5.42a.726.726 0 01-.728.722h-1.09a.364.364 0 01-.365-.36c0-.2.165-.362.364-.362h.91c.108 0 .18-.072.18-.18V1.403zM0 7.847V2.958c0-.359.337-.652.748-.652h4.114c.411 0 .749.293.749.652v4.89c0 .359-.338.652-.749.652H.748C.337 8.5 0 8.207 0 7.847z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
            </span>
          </span>
        </div>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--margin-top-medium StyledProps--color StyledProps--color-black"
      >
        delegates.delegateCreation.docker.scriptRunInShell
      </p>
      <div
        class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main"
      >
        <div
          class="StyledProps--font StyledProps--main verificationField StyledProps--intent-primary StyledProps--padding-small StyledProps--font-align-center StyledProps--flex"
        >
          <p
            class="StyledProps--font StyledProps--main StyledProps--margin-right-xxlarge StyledProps--font-small"
          >
            $ docker container exec -it &lt;container-id&gt; bash
          </p>
          <span
            class="bp3-popover-wrapper"
          >
            <span
              class="bp3-popover-target"
            >
              <div
                class=""
                tabindex="0"
              >
                <span
                  class="bp3-icon StyledProps--main copyIcon"
                  data-icon="copy-alt"
                >
                  <svg
                    fill="none"
                    height="20"
                    viewBox="0 0 8 9"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M7.273 1.403c0-.109-.073-.181-.182-.181H3.453c-.108 0-.181.072-.181.18v.181a.363.363 0 01-.727 0v-.361c0-.397.328-.722.728-.722h4c.4 0 .727.325.728.721v5.42a.726.726 0 01-.728.722h-1.09a.364.364 0 01-.365-.36c0-.2.165-.362.364-.362h.91c.108 0 .18-.072.18-.18V1.403zM0 7.847V2.958c0-.359.337-.652.748-.652h4.114c.411 0 .749.293.749.652v4.89c0 .359-.338.652-.749.652H.748C.337 8.5 0 8.207 0 7.847z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </div>
            </span>
          </span>
        </div>
      </div>
      <p
        class="StyledProps--font StyledProps--main StyledProps--margin-top-xlarge StyledProps--color StyledProps--color-black"
      >
        delegates.delegateCreation.docker.docLinkBefore
        <a
          href="https://ngdocs.harness.io/article/5ww21ewdt8-configure-delegate-proxy-settings"
          rel="noreferrer"
          target="_blank"
        >
          delegates.delegateCreation.docker.docLink
        </a>
      </p>
    </div>
  </div>
  <a
    class="hide"
    target="_blank"
  />
</div>
`;
